system/CodeIgniter.php

```diff
@@ -525,8 +527,8 @@ protected function handleRequest(?RouteCollectionInterface $routes, Cache $cache
             // Cache it without the performance metrics replaced
             // so that we can have live speed updates along the way.
             // Must be run after filters to preserve the Response headers.
-            if (static::$cacheTTL > 0) {
-                $this->pageCache->make($this->request, $this->response, static::$cacheTTL);
+            if ($this->pageCache->getTtl() > 0) {
+                $this->pageCache->make($this->request, $this->response);
             }
```

Comment:
Thank you for considering my suggestion, but in this snippet, I meant a slightly different approach.
```suggestion
            $this->pageCache->make($this->request, $this->response);
```
and then the decision on the need to cache the Response is made by the caching class.
```php
    public function make($request, ResponseInterface $response): bool
    {
        if ($this->ttl === 0) {
            return ; // bool (false or true)
        }
```
This is how I see the implementation, but I don't know if my approach is correct or not.

Reply:
If `$ttl` is 0, there is no need to cache it, so that seems a little better.

Done.